editor/src/clj/editor/resource_dialog.clj

```diff
@@ -25,19 +25,22 @@
             [editor.ui.fuzzy-choices :as fuzzy-choices]
             [editor.workspace :as workspace]
             [internal.graph.types :as gt]
-            [util.fn :as fn]))
+            [util.fn :as fn]
+            [util.thread-util :as thread-util]))
 
 (def ^:private fuzzy-resource-filter-fn (partial fuzzy-choices/filter-options resource/proj-path resource/proj-path))
 
 (defn- resource-node-ids->project-resources [basis resource-node-ids]
   (let [project-resources
         (into []
               (comp
+                (map thread-util/abortable-identity!)
```

Comment:
This throws an `InterruptedException` if the thread we're running on has been interrupted by someone cancelling the `future`, otherwise returns the value unaltered.